docs/documentation.md

```diff
@@ -0,0 +1,430 @@
+
+### Considerations
+
+1. ***Ghost listings*** - *Ghost listings are listings which don’t have an underlying NFT in the seller’s account. However, the listing is still available for buyers to attempt to purchase and which fails*. 
+
+    Ghost listings occur for two reasons: 
+
+    1. When a seller's NFT is sold in one marketplace but listings for that NFT in other marketplaces are not removed.
+    2. When the seller transfers out the listed NFT from the account that made the listings.
+
+    Usually, ghost listings will cause a purchaser’s transaction to fail, which is annoying but isn’t a significant problem. We recommend using the [`cleanupPurchasedListings`](#fun-cleanupPurchasedListings) function to mitigate the issues above. 
+
+    Ghost listings could be problematic for the seller if not cleaned up specifically when the listed NFT returns to the seller’s account after the original sale or transfer out. As a result the ghost listings would once again become able to facilitate purchases. This may be undesirable as the ghost listing price may be less than fair market value at the subsequent time.
+
+    ***Note:*** It may also be desirable for marketplaces or dApps to implement an off-chain notification service to inform users (eg: sellers) of listings that should be removed where the NFT for that listing no longer exists in the seller's account.
+
+2. ***Expired listings*** `NFTStorefrontV2` introduces a safety measure to flag an NFT listing as expired after a certain period. This can be set during listing creation to prevent the purchase through the listing after expiry has been reached. Once expiry has been reached the listing can no longer facilitate the purchase of the NFT. 
+
+    We recommend that using the [`cleanupExpiredListings`](#fun-cleanupExpiredListings) function to manage expired listings. 
+    
+    ***Note:*** We recommend that marketplaces and dApps filter out expired listings as they cannot be purchased.
+
+## Purchasing NFTs
+
+Purchasing NFTs through the `NFTStorefrontV2` is simple. The buyer has to provide the payment vault and the `commissionRecipient`, if applicable, during the purchase. The [`purchase`](#fun-purchase) API offered by the `Listing` facilitates the trade with the buyer in the seller's `Storefront`.
+
+During the listing purchase all `saleCuts` are paid automatically. This also includes distributing [royalties](#enabling-creator-royalties-for-nfts) for that NFT, if applicable. If the vault provided by the buyer lacks sufficient funds then the transaction will fail.
+
+### Considerations
+
+1. ***Auto cleanup*** `NFTStorefrontV2` provides the [`getDuplicateListingIDs`](#fun-getDuplicateListingIDs) function to support cleanup of duplicate listings. The only drawback of this applies to NFTs with large numbers of duplicate listings. When a NFT is listed in a large number of marketplaces and performs cleanup on sale, it will slow the purchase and in the worst case may trigger an out-of-gas error. 
```

Comment:
@satyamakgec one thing to check with you is whether I am referring to the right function here. Also, I was curious why this is in the purchase section. The main question I have is whether this is mainly best-practice guidance, rather than 'Considerations'?

Reply:
`getDuplicateListingIDs` doesn't do any cleanup it only returns the duplicate listing Ids for a given listing. `purchase` function internally calls the cleanup function that removes all the duplicate listings.

Technically the first statement is incorrect.